tests/vector/vecmemlayoutalign.cpp

```diff
@@ -0,0 +1,97 @@
+/* RUN: %{execute}%s
+
+   Testing vector element alignment, size and layout to see that it meets:
+    SYCL 1.2.1 Section 4.10.2.6 Memory layout and alignment
+*/
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <boost/test/minimal.hpp>
+
+using namespace cl::sycl;
+
+template <typename T>
+struct Matrix4x4 {
+  cl::sycl::vec<T, 4> data[4];
+};
+
+
+template <typename T>
+void TestVecType() {
+  Matrix4x4<T> Mt;
+
+  // We don't actually care what these values are just that they are consistent
+  // and checkable. Its A, B, C and D in the ASCII table, 65,66,67,68 in decimal
+  for (int i = 0; i < 4; ++i) {
+    Mt.data[i] = {0x41, 0x42, 0x43, 0x44};
+  }
+
+  cl::sycl::vec<T, 1> v1;
+  cl::sycl::vec<T, 2> v2;
+  cl::sycl::vec<T, 3> v3;
+  cl::sycl::vec<T, 4> v4;
+  cl::sycl::vec<T, 8> v8;
+
+  BOOST_CHECK(sizeof(v1) == 1 * sizeof(T));
+  BOOST_CHECK(sizeof(v2) == 2 * sizeof(T));
+  BOOST_CHECK(sizeof(v3) == 4 * sizeof(T));
+  BOOST_CHECK(sizeof(v4) == 4 * sizeof(T));
+  BOOST_CHECK(sizeof(v8) == 8 * sizeof(T));
+
+  BOOST_CHECK(sizeof(v1) == v1.get_size());
+  BOOST_CHECK(sizeof(v2) == v2.get_size());
+  BOOST_CHECK(sizeof(v3) == v3.get_size());
+  BOOST_CHECK(sizeof(v4) == v4.get_size());
+  BOOST_CHECK(sizeof(v8) == v8.get_size());
+
+  BOOST_CHECK(sizeof(Mt.data[0][0]) == sizeof(T));
+  BOOST_CHECK(sizeof(Mt.data[0]) == v4.get_size());
+  BOOST_CHECK(sizeof(Mt) == v4.get_size() * 4);
+
+  // is sequential and contigious in memory
+  for (int i = 0; i < 4; ++i)
+    BOOST_CHECK((static_cast<void*>(&v4[i]) ==
+                 static_cast<void*>(&v4[0] + i)));
+
+  // A magic trick! reinterpret_cast to a T* as the data should be contigious
```

Comment:
I am hesitating between contiguous or SYCL is a contagious illness... :-)
"contigious" is not in my spell-checker. Do you have a spell-checker by the way?

Reply:
Not at the moment, I've tried to integrate one into ATOM but it spellchecks everything including the code right now and for C++ it doesn't seem to have a filter to just check comments or at least it'll take a little more tinkering to get it to filter appropriately which I've been trying.